cpp/memilio/utils/miompi.cpp

```diff
@@ -0,0 +1,65 @@
+/* 
+* Copyright (C) 2023 German Aerospace Center (DLR-SC)
+*
+* Authors: Daniel Abele
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include "memilio/utils/miompi.h"
+
+#ifdef MEMILIO_ENABLE_MPI
+#include <mpi.h>
+#endif
+
+namespace mio
+{
+namespace mpi
+{
+Comm get_world()
```

Comment:
maybe. 

the communicator is used in the API of functions and classes, so there would be a lot of `ifdef MEMILIO_ENABLE_MPI` everywhere. I tried to limit the need for ifdef by providing dummy objects and functions for some MPI stuff. This function can be called everywhere, even if configured without MPI. rank and size rarely need to be used in APIs, they are usually only used when other MPI functions are used as well anyway. We could provide empty dummy functions for all of MPI so that we never need ifdef, but that's too much work.

Can also be added later when needed.